src/displayBackend/DisplayBackend.cpp

```diff
@@ -155,7 +154,14 @@ DisplayBackend::DisplayBackend(DisplayPtr display,
 	BackendBase("DisplBackend", deviceName),
 	mDisplay(display)
 {
-	mDisplay->start();
+	if(mDisplay)
+	{
+		mDisplay->start();
+	}
+	else
+	{
+		throw std::invalid_argument("DisplayBackend can not be constructed, because mDisplay is not initialized");
```

Comment:
Well, it is better move this check to the beginning of the constructor:
```
if (!mDisplay)
    throw ....
```
But, I'm not sure that we need this patch in the first place. This is not the library code, and according to logic, `display` should be always initialized.
I believe `ASSERT(display)` will be enough.

Reply:
fixed